scripts/hudson/narayana.sh

```diff
@@ -628,10 +628,13 @@ EOT
     rm -rf narayana-tomcat
     git clone https://github.com/jbosstm/narayana-tomcat.git
     echo "Executing Narayana Tomcat tests"
-    ./build.sh -f narayana-tomcat/pom.xml -fae -B -P${ARQ_PROF}-tomcat ${CODE_COVERAGE_ARGS} -Dtest.db.type=h2 -Dversion.org.jboss.narayana=${NARAYANA_CURRENT_VERSION} "$@" ${IPV6_OPTS} clean install "$@"
+    grep "<version.org.jboss.narayana>5.9.11.Final</version.org.jboss.narayana>" narayana-tomcat/pom.xml
+    [ $? -eq 0 ] || fatal "The version of Narayana in the jbosstm fork is not the expected version to replace"
+    sed -i "" "s#<version.org.jboss.narayana>5.9.11.Final</version.org.jboss.narayana>#<version.org.jboss.narayana>${NARAYANA_CURRENT_VERSION}</version.org.jboss.narayana>#g" narayana-tomcat/pom.xml
```

Comment:
I know the CI test run passed but I don't understand your sed syntax and it doesn't work on my Fedora release 34 laptop). Something like the following is what I'd expect:

`sed -i s/\<version\.org\.jboss\.narayana\>5.9.11.Final/\<version\.org\.jboss\.narayana\>${NARAYANA_CURRENT_VERSION}/g x`

I also didn't bother matching on the closing element.

Reply:
Regarding the initial `""` I see this being done here: https://github.com/jbosstm/narayana/blob/master/scripts/pre-release.sh#L63

I can't really remember why it was that way but maybe it was for MacOS?

I am not sure why that line isn't working on Fedora 34 - did you run it in isolation or did you run it as part of the script? Perhaps there is something else that has been set in the script that enables this kind of syntax?

Reply:
I did the following:

```bash
[mmusgrov@dev2 tmp]$ cat x
line1
<version.org.jboss.narayana>5.9.11.Final</version.org.jboss.narayana>
line2

[mmusgrov@dev2 tmp]$ echo ${NARAYANA_CURRENT_VERSION}
5.CURRENT
[mmusgrov@dev2 tmp]$ sed -i "" "s#<version.org.jboss.narayana>5.9.11.Final</version.org.jboss.narayana>#<version.org.jboss.narayana>${NARAYANA_CURRENT_VERSION}</version.org.jboss.narayana>#g" x
sed: can't read s#<version.org.jboss.narayana>5.9.11.Final</version.org.jboss.narayana>#<version.org.jboss.narayana>5.CURRENT</version.org.jboss.narayana>#g: No such file or directory
[mmusgrov@dev2 tmp]$ 
```

Reply:
I will research (later) what the alternate syntax is about.

Reply:
Even though it (appears to) break on my system, I will merge anyway after the test finishes. If does break on CI we would just push another update anyway.